packages/core/src/useForm/useForm.spec.ts

```diff
@@ -140,6 +140,101 @@ describe('form touched', () => {
     setFieldTouched('foo', false);
     expect(isTouched.value).toBe(false);
   });
+
+  test('sets touched state correctly for discriminated union paths', async () => {
+    const { setFieldTouched, isFieldTouched, setFieldValue, values } = await renderSetup(() => {
+      return useForm<any>({
+        initialValues: {
+          someConfig: {
+            nestedField1: 'value1',
+            nestedField2: 'value2',
+          },
+        },
+      });
+    });
+
+    /**
+     * Initialize touched state for all fields.
+     *
+     * This is necessary because in a normal form, fields are initialized
+     * internally through the `useFormField` hook, which is not used in this test.
+     * Order of operations is important here.
+     */
+    setFieldTouched('someConfig.nestedField1', false);
+    setFieldTouched('someConfig.nestedField2', false);
+    setFieldTouched('someConfig', false);
```

Comment:
@logaretm I'm not sure if there is a better way, but the initial touch state for fields is set internally by `useFormField`.

In the vitest, context here, fields don't get initialized naturally, so I'm having to simulate that aspect here in the same order that they would be initialized in real world.